docs/proposals/19-migrating-to-prometheus-operator.md

```diff
@@ -0,0 +1,544 @@
+
+Deploy the [prometheus-operator] and its CRDs. These components can be deployed
+via `ManagedResources`. The operator itself and some other components outlined
+in the GEP will be deployed in a new namespace called `monitoring`. The CRDs for
+the [prometheus-operator] and the operator itself can be found
+[here][prom-crd-bundle]. This step is a prerequisite for all other steps.
+
+### Shoot Monitoring
+
+Gardener will create a monitoring stack similar to the current one with the
+[prometheus-operator] custom resources.
+
+1. Most of the shoot monitoring is deployed via this
+    [chart][shoot-monitoring]. The goal is to create a similar stack, but not
+    necessarily with feature parity, using the [prometheus-operator].
+
+    - An example Prometheus object that would be deployed in a shoot's
+      control plane.
+
+    ```yaml
+    apiVersion: monitoring.coreos.com/v1
+    kind: Prometheus
+    metadata:
+      labels:
+        app: prometheus
+      name: prometheus
+      namespace: shoot--project--name
+    spec:
+      enableAdminAPI: false
+      logFormat: logfmt
+      logLevel: info
+      image: image:tag
+      paused: false
+      portName: web
+      replicas: 1
+      retention: 30d
+      routePrefix: /
+      serviceAccountName: prometheus
+      serviceMonitorNamespaceSelector:
+        matchExpressions:
+        - key: kubernetes.io/metadata.name
+          operator: In
+          values:
+          - monitoring
+          - shoot--project--name
+      podMonitorNamespaceSelector:
+        matchExpressions:
+        - key: kubernetes.io/metadata.name
+          operator: In
+          values:
+          - monitoring
+          - shoot--project--name
+      ruleNamespaceSelector:
+        matchExpressions:
+        - key: kubernetes.io/metadata.name
+          operator: In
+          values:
+          - monitoring
+          - shoot--project--name
+      serviceMonitorSelector:
+        matchLabels:
+          monitoring.gardener.cloud/monitoring-target: seed
+      podMonitorSelector:
+        matchLabels:
+          monitoring.gardener.cloud/monitoring-target: seed
+      storage:
+        volumeClaimTemplate:
+          spec:
+            accessModes:
+            - ReadWriteOnce
+            resources:
+              requests:
+                storage: 20Gi
+      version: v2.35.0
+    ```
+
+1. Contract between the shoot `Prometheus` and its configuration.
+
+    - `Prometheus` can discover `*Monitors` in different namespaces and also
+      by using labels.
+
+    - The monitoring configuration will be deployed in the namespace of the
+      shoot.
+
+    - In some cases, specific configuration is required (e.g. specific
+      configuration due to K8s versions). In this case, the configuration will
+      also be deployed in the shoot's namespace and Prometheus will also be able
+      to discover this configuration.
+
+    - To discover `ServiceMonitors` in the `monitoring` namespace and in the
```

Comment:
Now that we figured out that the `ignoreNamespaceSelectors` feature works differently than we assumed (the namespace of the Monitor is used for service discovery, and not the namespace of Prometheus), we have to adjust this bullet point as well: there will be no *Monitors in the `monitoring` namespace.

In addition to this: https://github.com/gardener/gardener/pull/6151/commits/29d4adb31250d22e7ffdf8fdf48c0b7fc3b96410